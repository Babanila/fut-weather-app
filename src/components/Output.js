import React from "react";
import { Link } from "react-router-dom";
import LazyLoad from "react-lazyload";
import Loading from "./Loading";
import { timeConverter } from "./Helpers";
import { OutputStyles } from "../styles/OutputStyles";

const Output = ({ data }) => {
    return data.map((item, i) => (
        <LazyLoad key={item.id} placeholder={<Loading />}>
            <Link to={`city/${item.id}`} key={item.id} id="link-cols">
                <OutputStyles className="card">
                    <picture className="card-picture">
                        <img src="https://wallpaperbro.com/img/438821.jpg" alt="weather" />
                    </picture>
                    <div className="card-content">
                        <div className="content-location">
                            <h3 data-setid="item-title">{item.properties.title}</h3>
                        </div>
                        <p className="card-section">
                            <span className="content-key">Type: </span>
                            <span className="content-value">{item.properties.type} </span>
                        </p>
                        <p>
                            <span className="content-key">Time: </span>
                            <span className="content-value">
                                {timeConverter(item.properties.time)}
                            </span>
                        </p>
                        <p>
                            <span className="content-key">Updated: </span>
                            <span className="content-value">
                                {timeConverter(item.properties.updated)}
                            </span>
                        </p>
                        <p>
                            <span className="content-key">RMS: </span>
                            <span className="content-value">{item.properties.rms} </span>
                        </p>
                        <p>
                            <span className="content-key">Dmin: </span>
                            <span className="content-value">
                                {item.properties.dmin || "Not available"}
                            </span>
                        </p>
                        <p>
                            <span className="content-key">MagType: </span>
                            <span className="content-value">{item.properties.magType} </span>
                        </p>
                    </div>
                </OutputStyles>
            </Link>
        </LazyLoad>
    ));
};

export default Output;
