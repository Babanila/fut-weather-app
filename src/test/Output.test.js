import React from "react";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import Output from "../components/Output";

describe("Output Component", () => {
    afterEach(cleanup);

    let weatherData;

    beforeEach(() => {
        weatherData = [
            {
                id: 1,
                properties: {
                    title: "sample-weather-data",
                    type: "earthquake",
                    time: 1574023724090,
                    updated: 1574024356540,
                    rms: 0.17999999999999999,
                    dmin: 0.1145,
                    magType: "ml",
                },
            },
            {
                id: 2,
                properties: {
                    title: "sample-data",
                    type: "earthquake",
                    time: 1574023725090,
                    updated: 1574084357540,
                    rms: 0.27999999999999999,
                    dmin: 0.2167,
                    magType: "ml",
                },
            },
        ];
    });

    it("should renders without crashing", () => {
        const OutputComponent = renderer.create(<Output data={weatherData} />).toJSON();
        expect(OutputComponent).toMatchSnapshot();
    });

    it("should render correct number of element in output component", () => {
        const OutputWrapper = shallow(<Output data={weatherData} />);
        expect(OutputWrapper.find("div").exists).toBeTruthy();
        expect(OutputWrapper.find("picture").exists).toBeTruthy();
        expect(OutputWrapper.find("img").exists).toBeTruthy();
        expect(OutputWrapper.find("span").exists).toBeTruthy();
        expect(OutputWrapper.find("p").exists).toBeTruthy();
        expect(OutputWrapper.find("p").length).toEqual(12);
        expect(OutputWrapper.find("li").length).toEqual(0);
    });
});
