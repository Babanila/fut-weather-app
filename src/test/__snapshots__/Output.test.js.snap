// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output Component should renders without crashing 1`] = `
Array [
  <div>
    <h5>
      Loading...
    </h5>
  </div>,
  <div>
    <h5>
      Loading...
    </h5>
  </div>,
]
`;
